vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+                                    d1start, d1count, dvar);
+        if (status != NC_NOERR) {
+            log_err("Error reading data from \"lat\" in %s",
+                    filenames.init_state);
+        }
+        for (i = 0; i < global_domain.n_ny; i++) {
+            if (dvar[i] != (double) global_domain.locations[i].latitude) {
+                log_err("Latitudes in initial state file do not "
+                        "match parameter file");
+            }
+        }
+        free(dvar);
+    }
+    else if (global_domain.info.n_coord_dims == 2) {
+        d2start[0] = 0;
+        d2start[1] = 0;
+        dvar = calloc(global_domain.n_ny * global_domain.n_nx, sizeof(*dvar));
+        d2count[0] = global_domain.n_ny;
+        d2count[1] = global_domain.n_nx;
+        status = nc_get_vara_double(nc.nc_id, lon_var_id,
+                                    d2start, d2count, dvar);
+        if (status != NC_NOERR) {
+            log_err("Error reading data from \"lon\" in %s",
+                    filenames.init_state);
+        }
+        for (i = 0; i < global_domain.n_ny * global_domain.n_nx; i++) {
+            if (dvar[i] != (double) global_domain.locations[i].longitude) {
+                log_err("Longitudes in initial state file do not "
+                        "match parameter file");
+            }
+        }
+        status = nc_get_vara_double(nc.nc_id, lat_var_id,
+                                    d2start, d2count, dvar);
+        if (status != NC_NOERR) {
+            log_err("Error reading data from \"lat\" in %s",
+                    filenames.init_state);
+        }
+        for (i = 0; i < global_domain.n_ny * global_domain.n_nx; i++) {
+            if (dvar[i] != (double) global_domain.locations[i].latitude) {
+                log_err("Latitudes in initial state file do not "
+                        "match parameter file");
+            }
+        }
+        free(dvar);
+    }
+    else {
+        log_err("global_domain.info.n_coord_dims should be 1 or 2");
+    }
+
+    // Variables for other dimensions
+    d1start[0] = 0;
+    d1count[0] = options.Nnode;
+
+    // soil thermal node deltas
+    dvar = calloc(options.Nnode, sizeof(*dvar));
+    get_nc_field_double(filenames.init_state, "dz_node",
+                        d1start, d1count, dvar);
+    for (i = 0; i < options.Nnode; i++) {
+        if (dvar[i] != soil_con[0].dz_node[i]) {
+            log_err("Soil node intervals in state file do not match "
+                    "those computed by VIC");
+        }
+    }
+    free(dvar);
+
+    // soil thermal node depths
+    dvar = calloc(options.Nnode, sizeof(*dvar));
+    get_nc_field_double(filenames.init_state, "node_depth",
+                        d1start, d1count, dvar);
+    for (i = 0; i < options.Nnode; i++) {
+        if (dvar[i] != soil_con[0].Zsum_node[i]) {
+            log_err("Soil node depths in state file do not match "
+                    "those computed by VIC");
+        }
+    }
```

Comment:
...but reading, writing and validating the mask is for a different/future PR, right?  (Just making sure)